### DISCUSSION TITLE
Apps - Create email templates and trigger emails.

### DISCUSSION THREAD

**Discussion by AlexBa** [shopware/platform#3506] [App FAQ]
### What kind of question do you have about the app system?

Templates

### Describe your question or feedback

Hello together,
my developers and I are often creating Shopware apps. After 2-3 years development, it seems like many missing features are there now. However, one open topic is still: emails.

We are missing the feature to create email template (in multiple languages) and a way to trigger them via app scripts (or other cool ways). This should be possible without creating an external service to avoid hosting fees and high availability issues.

Are there any plans to add this missing and important feature?

**Comment by OliverSkroblin** [comment]
Hi, 

thanks for the input. I recognize the need for this feature, and implementing a solution shouldn't be a problem.

The question that arises is how exactly to do it in detail. Creating new email templates is quite straightforward. The only question is, which APIs should we provide?

Currently, emails in our system are generated and sent through flows. In this context, it might be worth considering whether the app can simply trigger a configured flow to send an email.

Sending emails directly to the customer could be challenging (I believe), unless the email can be configured by the shop merchant. However, emails that don't go to the customer but to the merchant would be acceptable.

What APIs and their signatures did you have in mind?

**Reply by AlexBa** [reply]
Hello @OliverSkroblin ,
sry for the late reply.

A good way would be to add the email templates as files to the app. Something like MyApp/Resources/email-templates/de-DE/my-email.html.twig and MyApp/Resources/email-templates/de-DE/my-email.txt.twig. Maybe it could be good to reference them somehow in the manifest.xml to just tell Shopware that the files are existing.

Adding new flow actions and services in the app scripts would be important too to trigger them in more complex scenarios (something like: product stock changed => send a new email)

**Reply by OliverSkroblin** [reply]
Okay, I don't think adding mail templates via files should be a big challenge.
As far as sending is concerned, I would take a look at how we could implement `flow-trigger` via app scripts.
